app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,15 @@
+class CommentsController < ApplicationController
+  def create
+    @user = Post.find(params[:user_id])
+    @post = Post.find(params[:post_id])
+    @comment = Comment.new(params.require(:comment).permit(:text))
+    @comment.author = current_user
+    @comment.post = @post
+
+    if @comment.save
+      redirect_to user_post_path(@user, @post)
+    else
+      render :new
+    end
+  end
```

Comment:
[OPTIONAL]

- It would be awesome if you privately added a method for the comment parameters and passed them in like
`@comment = @current_user.comments.new(params)`Instead of that in line 5. You can write the method like this
```
  private 
  def comment_params
    params.require(:comment).permit(:text)
  end
```
- You can read more bout this [here](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides)